webpack/assets/javascripts/react_app/components/BreadcrumbBar/BreadcrumbSearch.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { DebounceInput } from 'react-debounce-input';
+
+class BreadcrumbSearch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: '',
+    };
+  }
+  render() {
+    const { onSearchChange, searchValue, timeout } = this.props;
```

Comment:
I think that inside this component we can call it `onChange` and `value` since there is nothing here except the search.
Consumer components can use the `search` context `<BreadcrumbSearch value={searchValue} onChange={onSearchChange} />`

Reply:
I would also prefer to change the `timeout` to `debounceTimeout` to make it more clear.